py3status/modules/dpms.py

```diff
@@ -52,26 +56,29 @@ def dpms(self):
         """
         Display a colorful state of DPMS.
         """
-        self.run = system('xset -q | grep -iq "DPMS is enabled"') == 0
-        icon = self.icon_on if self.run else self.icon_off
+        run = system('xset -q | grep -iq "DPMS is Enabled"') == 0
+        icon = self.icon_on if run else self.icon_off
 
         return {
+            'cached_until': self.py3.time_in(self.cache_timeout),
             'full_text': self.py3.safe_format(self.format, {'icon': icon}),
-            'color': self.py3.COLOR_ON or self.py3.COLOR_GOOD if self.run
+            'color': self.py3.COLOR_ON or self.py3.COLOR_GOOD if run
             else self.py3.COLOR_OFF or self.py3.COLOR_BAD
         }
 
     def on_click(self, event):
         """
-        Enable/Disable DPMS on left click.
+        Enable/Disable DPMS on click.
         """
-        if event['button'] == 1:
-            if self.run:
-                self.run = False
-                system("xset -dpms;xset s off")
+        run = system('xset -q | grep -iq "DPMS is Enabled"') == 0
+
+        if event['button'] == self.button_toggle:
+            if run:
+                run = False
+                self.py3.command_run("xset -dpms s off")
             else:
-                self.run = True
-                system("xset +dpms;xset s on")
+                run = True
+                self.py3.command_run("xset +dpms s on")
```

Comment:
If we are being picky then we only need to check if we are running when we toggle and the `run = True\False` lines are not needed.